contracts/KyberNetwork.sol

```diff
@@ -347,12 +278,10 @@ contract KyberNetwork is Withdrawable, Utils {
         uint minConversionRate,
         address walletId
     )
-        internal
+        public
+        payable
         returns(uint)
     {
-        require(tx.gasprice <= maxGasPrice);
-        require(validateTradeInput(src, srcAmount, destAddress));
-
         Amount memory amount;
```

Comment:
do you really need a struct here?

Reply:
yes. 
I get stack too deep compilation error.
u think we can remove any parameter.??